packages/gatsby/src/datastore/lmdb/lmdb-datastore.ts

```diff
@@ -237,10 +240,31 @@ function updateDataStore(action: ActionsUnion): void {
       clearIndexes()
       break
     }
+    case `ADD_CHILD_NODE_TO_PARENT_NODE`: {
+      let fn
+      const dbs = getDatabases()
+      const key = action.payload.id + action.payload.internal.type
+      if (!debounceFunctionsPerNodeType.has(key)) {
+        fn = _.debounce(
+          _action => {
+            updateNodes(dbs.nodes, _action)
+            updateNodesByType(dbs.nodesByType, _action)
+          },
+          100,
+          { leading: true, trailing: true }
+        )
```

Comment:
Can this potentially lose information when 2 different plugins try to add child nodes to same parent node around the same time? `key` would be the same (as it's calculated from just parent node)

Reply:
Other issue - because it's time based, with unlucky timing we might finish sourcing before final debounced writes happen now. This might cause unforseen problems because LMDB might not be fully complete - if we want to batch things, we should ensure to call `fn.flush()` on all pending ones before sourcing finish